doc/architectural/compilation-and-development.md

```diff
@@ -270,3 +270,28 @@ branch, e.g. to run it on lines that have been changed from `develop`:
 
 There are also instructions for adding this as a git pre-commit hook in
 [CODING_STANDARD.md](https://github.com/diffblue/cbmc/blob/develop/CODING_STANDARD.md#pre-commit-hook-to-run-cpplint-locally).
+
+
+\section compilation-and-development-section-time-profiling Time profiling
+
+To do time profiling with a tool like `gprof`, the flags `-g` and `-pg` must be
+used which compiling, and `-pg` must be used when linking. If you are building
+with cmake you can just add "-Denable_profiling=1" to your cmake invocation, and
+reload cmake before building your desired binary.
+
+Run your binary as normal. A file called gmon.out will be created in the
+working directory of the binary at the end of execution. In most instances this
+will be the same as the working directory at the beginning of execution. It is
+also possible to choose the output location using an environment variable -
+search online for more details.
+
+Make sure gprof is installed by running `gprof -v`. If it is not installed then
+run `sudo apt install binutils`.
+
+Run `gprof <path-to-binary> <path-to-gmon.out>` and redirect the output to a
+file. This will take a while to run.
```

Comment:
:bulb: In case you want to be more specific than "a while", I ran jbmc without options on a very simple Java main method (one assignment to an int, one assert on the int) and this step took 12min for me. Of course this will depend on the machine this is run on but it would be good to document a general order of magnitude of how long this could take.